leveldb_backend.go

```diff
@@ -0,0 +1,48 @@
+package gocelery
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/syndtr/goleveldb/leveldb"
+)
+
+// levelDBBackend implements the CeleryBackend based on levelDB
+type levelDBBackend struct {
+	db *leveldb.DB
+}
+
+// NewLevelDBBackend returns an levelDB implementation of CeleryBackend
+func NewLevelDBBackend(db *leveldb.DB) CeleryBackend {
+	return levelDBBackend{db: db}
+}
+
+// GetResult returns the result stored at key
+func (ldb levelDBBackend) GetResult(key string) (*ResultMessage, error) {
+	data, err := ldb.db.Get([]byte(fmt.Sprintf("celery-leveldb-result-%s", key)), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	res := new(ResultMessage)
+	err = json.Unmarshal(data, res)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal result: %v", err)
+	}
+
+	return res, nil
+}
+
+// SetResult stores the result at key taskID
+func (ldb levelDBBackend) SetResult(taskID string, result *ResultMessage) error {
```

Comment:
Or use in both places `taskID` (gocelery naming) and change the prefix to `celery-task-meta-`

In addition, if you could create a constant with that prefix would be great.

Reply:
Every backend should have its own prefixes and I think they should not be set as constants so that we could avoid re-use by others in different backends